### DISCUSSION TITLE
Audit Log Streaming to Datadog -  Beta Feedback

### DISCUSSION THREAD

**Discussion by boylejj** [community/community#21347] [Enterprise]
The purpose of this discussion is to collect feedback from Enterprises participating in the private beta for configuring audit log streaming to a Datadog endpoint. GHEC administrators interested in participating in the private beta should reach out to your GitHub account manager or [contact our sales team](https://github.com/enterprise/contact) to make the feature available for your enterprise.

### Setting up streaming to Datadog

To set up streaming to Datadog, you must create a client token or an  API key in Datadog, then configure audit log streaming in GitHub Enterprise Cloud using the token for authentication. You do not need to create a bucket or other storage container in Datadog.

After you set up streaming to Datadog, you can see your audit log data by filtering by "github.audit.streaming." For more information, see [Log Management](https://docs.datadoghq.com/logs/).

1. If you don't already have a Datadog account, create one.

1. In Datadog, generate a client token or an API key, then click **Copy key**. For more information, see [API and Application Keys](https://docs.datadoghq.com/account_management/api-app-keys/) in Datadog Docs.

1. In the top-right corner of GitHub.com, click your profile photo, then click **Your enterprises**.

1. In the list of enterprises, click the enterprise you want to view.

1. In the enterprise account sidebar, click **Settings**.

1. Under " Settings", click **Audit log**.

1. Under "Audit log", click Log streaming.

1. Select the **Configure stream** dropdown menu and click **Datadog**.

1. Under "Token", paste the token  you copied earlier.

1. Select the "Site" dropdown menu and click your Datadog site. To determine your Datadog site, compare your Datadog URL to the table in [Datadog sites](https://docs.datadoghq.com/getting_started/site/) in Datadog Docs.

1. To verify that GitHub can connect and write to the Datadog endpoint, click **Check endpoint**.

1. After you have successfully verified the endpoint, click **Save**.

1. After a few minutes, confirm that audit log data is appearing on the **Logs** tab in Datadog. If audit log data is not appearing, confirm that your token and site are correct in GitHub.

**Comment by cardoppler** [comment]
Hi @boylejj ! I have followed the steps above but no luck with receiving logs in DD. What's the best way to get support; here or via email? I am not sure if posting screenshots in this thread is the most appropriate thing to do 😅 (although it might help others!)

**Reply by boylejj** [reply]
@cardoppler, feel free to send an email summarizing what you're seeing to me and @maxprit. We'll work through the challenges you're seeing, and adjust the instructions as appropriate.

**Reply by avgalani** [reply]
hi @boylejj, we're also encountering the same issue.

We've followed the instructions to the letter and by all means it should work, however we don't see any logs popping into Datadog. 
We were sure to select the correct region for our Datadog instance (EU1), as well as to use a valid API key.

**Reply by boylejj** [reply]
@avgalani, we believe we found the cause of the issue and have shipped a fix. Please try again and let me know if you're still experiencing issues.

**Comment by greta-ramaneckaite** [comment]
Hi @boylejj,
I've encountered a strange bug inside the logs. I'm working on setting up some Detection Rules for monitoring public repository creation and I spotted that some logs contain actor's location:
If an actor modifies a repositories access from public to private - there is a location in the log. But when an actor creates a public repository or modifies it from private to public - there is no location. For our configuration - we would like to see the opposite instead.
Is it possible to toggle this or maybe enable the location specification for all of the logs? 

Example section from a log:
`
actor_location { 
  country_code: RO
}
`

**Reply by boylejj** [reply]
@greta-ramaneckaite ... Thanks for the feedback. What you're seeing is currently the expected behavior, as any activity tied to public repositories will not have location related information (IP address or actor_location) included in the audit log entry.

That being said, I can certainly understand why this information would be very important when a private repository is made public. Particularly, determining if this was done by mistake by someone or done maliciously by a bad actor. I will get this written up and loaded into our backlog for consideration and prioritization.

**Reply by greta-ramaneckaite** [reply]
Thanks @boylejj !

**Comment by PeterFagans** [comment]
So I was following the above instructions but I don't seem to have a drop down box per section 8:

<img width="1900" alt="Screen Shot 2022-08-24 at 5 25 32 PM" src="https://user-images.githubusercontent.com/86617616/186526741-99976cb0-72e8-4c53-8f24-e85259695e8d.png">

 Is it because I already have a stream setup?

**Reply by cardoppler** [reply]
Hi @PeterFagans - Yes, I had the same issue, it looks like it's not possible to have both active at the same time. Only after deleting the existing old stream (to S3), I could see the new option to setup "Streaming to DataDog".

<img width="1145" alt="image" src="https://user-images.githubusercontent.com/1302027/186531770-b3be5cd0-9b55-4d4e-968d-a3d27b9bea66.png">

**Reply by PeterFagans** [reply]
Thank you! Thats what I thought but wanted confirmation before I blew away the S3 connector

**Reply by boylejj** [reply]
@PeterFagans and @cardoppler.... How helpful would it be to be able to configure multiple streams? I can see value in being able to have a `test` and `prod` for streams to that same endpoint. But, are there instances where you would like to stream the same enterprise audit log to two different endpoints?

**Comment by PeterFagans** [comment]
I could see a situation where you want to additionally stream the logs to a
secure place for archival purposes where you are confident that nothing was
manipulated or for backup.
Also it might be handy to be able to parse the stream ahead of time so you
could direct the stream to various endpoints by repo.

Peter

**Comment by ozonni** [comment]
why is this feature only available for enterprises account and not just for account on enterprises pricing plan?

**Comment by dpihlaja** [comment]
Not sure if this product is still in Beta or not - but I'm trying to track down where the maintainers live for the github->dd audit log streaming integration for an issue with the integration @boylejj (your github profile makes it seem like you are still the PM of this product)

The integration dashboard in Datadog is broken so will throw down some details here. [I started a Slack thread in the Datadog Public Integrations channel here](https://datadoghq.slack.com/archives/C3ZGCF88Z/p1712266619883059) which is just me talking to myself (not expecting any official responses there, just dunno what support department to hit up here) 

Dashboard widgets are currently using this attribute in the queries: 
```
@github.event
```

However these are the attributes actually on the current logs: `
```
@action:git.fetch
@evt.action:git.fetch
```

To fix this, can either change the widget to pick one of the existing attributes, or if nothing else is using those facets, change the pipeline processor to remap to `@github.event` instead.

(sidenote; why bother preserving the parent attributes here.... some legacy backwards compatibility thing? This creates larger logs with duplicate information that creates multiple ways of doing the same thing, which leads to multiple flows to maintain)

<img width="834" alt="Screenshot 2024-04-04 at 4 32 04 PM" src="https://github.com/community/community/assets/46490886/e9c5bfe4-dee3-458b-84b6-2539ab9bd380">
<img width="542" alt="Screenshot 2024-04-04 at 4 23 16 PM" src="https://github.com/community/community/assets/46490886/4d01e681-69f5-4f03-8e7e-94028ca044dd">
<img width="681" alt="Screenshot 2024-04-04 at 4 21 24 PM" src="https://github.com/community/community/assets/46490886/c89fb3b2-3ca4-4aea-a256-2bd87d201f15">
